### PR TITLE
Allows for limited-time supply buffers

### PR DESCRIPTION
**Pull request by sonatav2**
Introduces a `supply_buffer_length` parameter which can be set to the maximum time step up to which you would like to include a supply buffer. This was developed for scenarios such as the eg01-eg30 transition scenario, where waiting to deploy new reactors until the previous reactors begin decommissioning (causing the power supply to fall) results in a period of undersupply during the transition. This undersupply can be minimized by using a combination of facility_preference and a limited-time buffer to deploy new reactors shortly before supply falls below demand.

Example: If the initial facilities will begin decommissioning at time step 950, a combination of facility_preference `(t-940)` and a 30% power supply buffer with `supply_buffer_length = 950` can be implemented to trigger an initial deployment of reactors at time step 941. 

Without `supply_buffer_length`, the supply buffer would continue to effect the demand equation through the end of the simulation, and there would be an overdeployment of reactors. This allows the user to trigger pre-transition deployment without permanently effecting the demand for the commodity.

### CODE REVIEW

File: tests/integration_tests/buffer_test.py [hunk 1]
```diff
@@ -722,3 +722,192 @@ def test_capacity_buffer_num():
             count += 1
 
     assert(count == 0)
+
+
+# -------------------------------------------------------------------------- #
```
Comment (katyhuff) [inline tests/integration_tests/buffer_test.py:727]: I prefer not having this kind of variable defined at the global scope. Ideally, this template would be in a setup function (see http://doc.pytest.org/en/latest/fixture.html ) or if it's only used in one test, put it in the test function itself.

File: tests/integration_tests/buffer_test.py [hunk 2]
```diff
@@ -722,3 +722,192 @@ def test_capacity_buffer_num():
...
+                "basis": "mass",
+                "name": "spent_uox",
+                "nuclide": [{"comp": "50", "id": "Kr85"},
+                            {"comp": "50", "id": "Cs137"}]
+            }
+        ]
+    }
+}
+
+nobuf_template_length = copy.deepcopy(TEMPLATE_length)
```
Comment (katyhuff) [inline tests/integration_tests/buffer_test.py]: same as above comment regarding fixtures.

File: tests/integration_tests/buffer_test.py [hunk 3]
```diff
@@ -722,3 +722,192 @@ def test_capacity_buffer_num():
...
+            }
+        },
+        "name": "source_inst"
+    },
+    "name": "SingleRegion"
+}
+})
+
+
+yesbuf_template_length = copy.deepcopy(TEMPLATE_length)
```
Comment (katyhuff) [inline tests/integration_tests/buffer_test.py]: same as above comment regarding fixtures.

**Review by katyhuff — CHANGES_REQUESTED** [review]
looks good. one minor comment about global scope in the tests.

**Comment by katyhuff** [comment]
@robfairh @gwenchee @FlanFlanagan @sonatav2 

Can someone bring these improvements in and bring this PR to a close?

**Comment by FlanFlanagan** [comment]
@katyhuff 

It looks like @sonatav2 followed the procedure from the other TEMPLATE in the same document. Do you want that updated as well?

**Comment by FlanFlanagan** [comment]
I'll be taking a crack at this! Thanks!

**Comment by FlanFlanagan** [comment]
I made changes to this file, but I don't have the ability to push to her branch? Should I make a new PR?

**Comment by robfairh** [comment]
Yes, do it @FlanFlanagan

**Comment by katyhuff** [comment]
This seems good -- @FlanFlanagan @sonatav2 Since this one is passing the tests, is it superior to #290 .

**Comment by FlanFlanagan** [comment]
This is superior! I'll close mine!

**Comment by robfairh** [comment]
This version was never fully implemented and it is no longer needed.

**Pull request closed without merging**